### PR TITLE
Space Traveler's Hub Project

### PR DESCRIPTION
**Pull request by HFG43**
Hi there,
we are sending this pull request for you to please review our code. Thanks in advance for your feedback!! 😎😎🙌

@AndreaM2429 && @HFG43 
For this project, we have worked on pair programming and have done the following:
## **Milestones:**

- [ ] Install React Redux, Redux Logger, and React Router. ✔
- [ ] Create routes and view components for: rockets, missions, and my profile.✔
- [ ] Use `<NavLink />` for the page navigation links and style active class to indicate which section/page user is currently on (underline active navigation link). ✔
- [ ] Fetch data from the SpaceX API for Rockets & Missions info✔
- [ ] Create `actions` and `reducers` to reserve Rockets & join Missions ✔
- [ ] Create `actions` and `reducers` to cancel Rockets & leave Missions ✔
- [ ] `Actions` and `reducers` to render conditional components ✔
- [ ] Use `filter()` to render a list of all reserved rockets & all joined missions ✔
- [ ] Write Unit tests ✔

## **General Requirements:**

- [ ] No linters errors
- [ ] GitFlow
- [ ] Best practices on JavaScript
- [ ] Document in a professional way

### CODE REVIEW

File: src/Redux/missions/MissionsSlice.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  const missionsList = [];
+  missions.forEach((mission) => {
+    missionsList.push({
+      mission_id: mission.mission_id,
+      mission_name: mission.mission_name,
+      description: mission.description,
+    });
+    return missionsList;
+  });
+  return missionsList;
```
Comment (AbayomiOlaoye) [inline src/Redux/missions/MissionsSlice.js]: - [ ] Great job with your team's project but I think you need to check how you update your store to ensure things are not buggy. If a mission or a rocket is booked, once the browser refreshes, the update is gone and back to default -as shown below:
 
Studio_Project_V1

  - Kindly find a way to fix this anomaly on both your mission & rockets

**Review by AbayomiOlaoye — CHANGES_REQUESTED** [review]
## STATUS: CHANGE REQUIRED ♻️ 

Hi @Team :wave: :wave: :clap: , 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

so_close

## Highlights
- Descriptive Pull Request message :heavy_check_mark: 
- Good use of GitHub flow :heavy_check_mark: 
- Meaningful commit messages :heavy_check_mark: 
- No linter error :heavy_check_mark: 
- Descriptive ReadMe documentation :heavy_check_mark: 
- All tests are passing :heavy_check_mark: 

## Suggested changes
- [ ] _Check the comments under the review._

Please feel free to use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- You might also consider making it more mobile-friendly

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/Redux/missions/MissionsSlice.js (hunk 1 above)
Comment (AndreaM2429) [reply src/Redux/missions/MissionsSlice.js]: Hello, @AbayomiOlaoye I appreciate your time to review the project but I want to clarify that the requirements said that we don't have to preserve the data on the browser as you mentioned.
image
[Here](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md) is the link to the requirements.

**Comment by AbayomiOlaoye** [comment]
Oh, great idea but you know, your store is not living in some browser but
an API where you make POST and GET requests to. So, it should be up to date
and should prove useable to users.

I don't know if you get my point now

**Comment by AndreaM2429** [comment]
Hi, @AbayomiOlaoye I got your point about the data not leaving in the browser but the requirements are very clear about **Do not preserve user data**.
So it is not mandatory to make a POST request.

File: src/Redux/missions/MissionsSlice.js (hunk 1 above)
Comment (peterrobert) [reply src/Redux/missions/MissionsSlice.js]: @AbayomiOlaoye I have to agree with @AndreaM2429. This change is not a requirement. What is important is the fact that if you navigate between the pages, the state is maintained. However, in order to have that maintained state when the window reloads, the data should have persisted somewhere like in a database or local storage with that given change to be maintained.

**Review by peterrobert — APPROVED** [review]
Hi @HFG43 and @AndreaM2429 , 


### Highlights!

[ ] Good feature branch name ✔️
[ ] Descriptive pull request ✔️


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**